### DISCUSSION TITLE
recommended install for ubuntu 18.04 not working with /home/$USER/.local/bin pip/jupyter present

### DISCUSSION THREAD

**Discussion by LukeAI** [pyenv/pyenv#1974] [Q&A] [answered]
I have the latest pyenv and pyenv-virtualenv installed and sourced in the way described in the READMEs

The problem is that, by default if you have a version of pip (or jupyter) installed in /home/luke/.local/bin then it will end up taking precedence over the pyenv shims so there isn't a convenient way to access an env's pip.

My question is: Is there some neat non-hacky solution to this? I guess I could manually reinsert `home/luke/.pyenv/shims` to the front of PATH in .bashrc but I feel like I'm not supposed to?

```
echo $PATH
/home/luke/.pyenv/plugins/pyenv-virtualenv/shims:/home/luke/.local/bin:/home/luke/.pyenv/shims:/home/luke/.pyenv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

```
type -a pip
pip is /home/luke/.local/bin/pip
pip is /home/luke/.pyenv/shims/pip
```

```
~/.bashrc
*all the default stuff*
# pyenv
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"
```

```
~/.profile
export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init --path)"

# if running bash
if [ -n "$BASH_VERSION" ]; then
    # include .bashrc if it exists
    if [ -f "$HOME/.bashrc" ]; then
	. "$HOME/.bashrc"
    fi
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/bin" ] ; then
    PATH="$HOME/bin:$PATH"
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
_byobu_sourced=1 . /usr/bin/byobu-launch 2>/dev/null || true

```

**Comment by native-api** [comment, marked as answer]
A solution would be to add `eval "$(pyenv init --path)"` to the end of `~/.profile` in the Ubuntu setup (the other commands need to remain before the part that sources `~/.bashrc` because `eval "$(pyenv init -)"` needs them).

**Reply by LukeAI** [reply]
thankyou, that works! Can I suggest that this be the recommended setup in the README.md? This must be quite common - I'm happy to amend myself if that would be welcome?

**Reply by native-api** [reply]
Sure. Could you make a PR?

**Reply by LukeAI** [reply]
sure, will do.
